domain/lib/src/model/sharedspacedocument/work_group_document.dart

```diff
@@ -0,0 +1,87 @@
+// suite, developed by Linagora.
+//
+// Copyright (C) 2020 LINAGORA
+//
+// This program is free software: you can redistribute it and/or modify it under the
+// terms of the GNU Affero General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later version,
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:domain/domain.dart';
+import 'package:http_parser/http_parser.dart';
+
+class WorkGroupDocument extends WorkGroupNode {
+  final int size;
+  final MediaType mediaType;
+  final bool hasThumbnail;
+  final DateTime uploadDate;
+  final bool hasRevision;
+  final String sha256sum;
+
+  WorkGroupDocument(
+    WorkGroupNodeId workGroupNodeId,
+    WorkGroupNodeId parentWorkGroupNodeId,
+    WorkGroupNodeType type,
+    SharedSpaceId sharedSpaceId,
+    DateTime creationDate,
+    DateTime modificationDate,
+    String description,
+    String name,
+    Account lastAuthor,
+    this.size,
+    this.mediaType,
+    this.hasThumbnail,
+    this.uploadDate,
+    this.hasRevision,
+    this.sha256sum) : super(
+      workGroupNodeId,
+      parentWorkGroupNodeId,
+      type,
+      sharedSpaceId,
+      creationDate,
+      modificationDate,
+      description,
+      name,
+      lastAuthor);
+
+  @override
+  List<Object> get props => [
```

Comment:
[Confirm] As we discussed before, can we try to implement partial `props` at `WorkGroupNode` and `WorkGroupDocument` can re-use of it's parent `props`

If oki, we also can apply for Dto classes

Reply:
yeah, forgot this. let me try now

Reply:
work like a charm

Reply:
I think, WorkGroupDocument has override WorkGroupNode. So I think it's `props` should be extend from it's parent;
```suggestion
List<Object> get props => [
    ...super.props,
    size,
    hasThumbnail,
    uploadDate,
    hasRevision,
    sha256sum,
];
```

About the `spread operator (...)`. We can read more at [here](https://github.com/dart-lang/language/blob/master/accepted/2.3/spread-collections/feature-specification.md)